src/gluonts/model/lstnet/_network.py

```diff
@@ -0,0 +1,329 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Optional
+import warnings
+
+# Third-party imports
+import mxnet as mx
+from mxnet.gluon import nn, loss
+
+# First-party imports
+from gluonts.model.common import Tensor
+from gluonts.core.component import validated, DType
+from gluonts.block.scaler import MeanScaler, NOPScaler
+
+
+class LSTNetBase(nn.HybridBlock):
+    @validated()
+    def __init__(
+        self,
+        num_series: int,
+        channels: int,
+        kernel_size: int,
+        rnn_cell_type: str,
+        rnn_num_layers: int,
+        skip_rnn_cell_type: str,
+        skip_rnn_num_layers: int,
+        skip_size: int,
+        ar_window: int,
+        context_length: int,
+        horizon: Optional[int],
+        prediction_length: Optional[int],
+        dropout_rate: float,
+        output_activation: Optional[str],
+        scaling: bool,
+        dtype: DType,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.num_series = num_series
+        self.channels = channels
+        assert (
+            channels % skip_size == 0
+        ), "number of conv1d `channels` must be divisible by the `skip_size`"
+        self.skip_size = skip_size
+        assert (
+            ar_window > 0
+        ), "auto-regressive window must be a positive integer"
+        self.ar_window = ar_window
+        if not ((horizon is None) ^ (prediction_length is None)):
+            raise ValueError(
+                "Only one of `horizon` and `prediction_length` can be used at a time"
+            )
+        if horizon is not None:
+            assert horizon > 0, "`horizon` must be greater than zero"
+            self.horizon = horizon
+            self.prediction_length = None
+        if prediction_length is not None:
+            assert (
+                prediction_length > 0
+            ), "`prediction_length` must be greater than zero"
+            self.prediction_length = prediction_length
+            self.horizon = None
```

Comment:
The `^` is bitwise xor, I suggest replacing these assertions with the following (please double check)

```suggestion
        assert (
            not ((horizon is None) == (prediction_length is None))
        ), "Exactly one of `horizon` and `prediction_length` must be set"
        assert horizon is None or horizon > 0, "`horizon` must be greater than zero"
        assert (
            prediction_length is None or prediction_length > 0
        ), "`prediction_length` must be greater than zero"
        self.prediction_length = prediction_length
        self.horizon = horizon
```